task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,184 @@
+    tekton.dev/tags: "konflux"
+  name: sast-shell-check
+spec:
+  description: >-
+    ShellCheck is a static analysis tool, gives warnings and suggestions for bash/sh shell scripts.
+  results:
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+  params:
+    - name: image-url
+      description: Image URL.
+      type: string
+      default: ""
+    - name: image-digest
+      description: Image digest to report findings for.
+      type: string
+      default: ""
+    - name: KFP_GIT_URL
+      description: git repository to download known false positives files from
+      type: string
+      # FIXME: Red Hat internal projects will default to https://gitlab.cee.redhat.com/osh/known-false-positives.git
+      # when KONFLUX-4530 is resolved
+      default: ""
+    - name: PROJECT_NVR
+      description: Name-Version-Release (NVR) of the scanned project, used to find path exclusions
+      type: string
+      default: ""
+    - name: RECORD_EXCLUDED
+      type: string
+      description: |
+        Whether to record the excluded findings (default to false).
+        If `true`, the excluded findings will be stored in `excluded-findings.json`.
+      default: "false"
+    - name: IMP_FINDINGS_ONLY
+      type: string
+      description: Whether to include important findings only
+      default: "true"
+  steps:
+    - name: sast-shell-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.7@sha256:cf6808a3bd605630a5d9f20595ff7c43f8645c00381219d32f5a11e88fe37072
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+      script: |
+        #!/usr/bin/env bash
+        set -x
+        . /utils.sh
+
+        # FIXME: once https://github.com/konflux-ci/konflux-test/pull/303 is available in latest image,
+        # work around below is no longer needed
+        microdnf install -y ShellCheck
+        microdnf download csmock-plugin-shellcheck && rpm -ivh --nodeps ./csmock-plugin-shellcheck-*.rpm && rm -f ./csmock-plugin-shellcheck-*.rpm
+
+        OUTPUT_FILE="shellcheck-results.json"
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)/source
+
+        # generate all shellcheck result JSON files to $SC_RESULTS_DIR, which defaults to ./shellcheck-results/
+        /usr/share/csmock/scripts/run-shellcheck.sh "$SOURCE_CODE_DIR"
+
+        CSGREP_OPTS=(
+            --mode=json
+            --strip-path-prefix="$SOURCE_CODE_DIR"/
+            --remove-duplicates
+            --embed-context=3
+        )
```

Comment:
We should also record the revision of the used scanner in scan properties via csgrep --set-scan-prop in a similar way as we record the revision of KFP: https://github.com/csutils/csdiff/pull/207/files

Reply:
the scan prop in the sarif is like this:
```
  "inlineExternalProperties": [
    {
      "externalizedProperties": {
        "csmock-plugin-shellcheck": "csmock-plugin-shellcheck-3.7.0-1.el9"
      }
    }
  ],
```

Reply:
Once https://github.com/konflux-ci/konflux-test/pull/303 is resolved (and the resulting image used by this task), the version should be given by the base image.  Nevertheless, I think we should rather record the version of `ShellCheck` itself, not the helper script.

Reply:
Sure. I've updated it, now the sarif excerpt looks like this:
{code}
$ cat shellcheck-results.sarif | jq '.inlineExternalProperties'
[
  {
    "externalizedProperties": {
      "ShellCheck": "ShellCheck-0.8.0-2.el9"
    }
  }
]
{code}